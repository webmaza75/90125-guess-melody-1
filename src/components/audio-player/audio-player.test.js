import React from 'react';
import renderer from 'react-test-renderer';

import AudioPlayer from './audio-player.jsx';
import question from '../../mocks/genre-question.js';

window.HTMLMediaElement.prototype.play = () => {};
window.HTMLMediaElement.prototype.pause = () => {};

it(`AudioPlayer is rendered correctly`, () => {
  const onPlayButtonClick = jest.fn();
  const {answers} = question;
  const tree = renderer.create(<AudioPlayer
    src={answers[0].src}
    isPlaying={true}
    onPlayButtonClick={onPlayButtonClick}
  />).toJSON();

  expect(tree).toMatchSnapshot();

});
