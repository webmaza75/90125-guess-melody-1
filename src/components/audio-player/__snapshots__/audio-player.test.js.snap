// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AudioPlayer is rendered correctly 1`] = `
<div
  className="game__track"
>
  <button
    className="track__button track__button--pause"
    disabled={true}
    onClick={[Function]}
    type="button"
  />
  <div
    className="track__status"
  >
    <audio />
    <div
      className="track__progress"
      style={
        Object {
          "background": "#707070",
          "height": "2px",
          "marginTop": "55px",
          "width": "0%",
        }
      }
    />
  </div>
</div>
`;
